pkg/build/pipelines/git-checkout.yaml

```diff
@@ -187,7 +187,7 @@ pipeline:
 
           vr cd "$workdir"
           msg "tar -c . | tar -C \"$dest_fullpath\" -x"
-          ( tar -c . ; echo $? > "$rcfile") | tar -C "$dest_fullpath" -x
+          ( tar -c . ; echo $? > "$rcfile") | tar -C "$dest_fullpath" -x --no-same-owner
```

Comment:
Because we're using Busybox, `--no-overwrite-dir` is not supported.

Reply:
We had to install `gnutar` for the recent xattr work, maybe it makes sense to install it regardless of runner?

Reply:
I'm a bit skeptical about introducing a further change to this pipeline as it's used across all Wolfi packages, but I don't have a strong opinion on that and I'm not against it.

Reply:
I'm ok for the switch to gnutar, it's posix compliant, so it won't break current compatibility, but opens up to more modern features